Delete temp files after demo primary and secondary run

shikhar394 opened the pull request:
Fix for issue #24 
Cleans up temp folders and files after demos run.

shikhar394 committed:
Demos with atexit functionality to remove temp folders after scripts end

shikhar394 committed:
Initialized atexit() with global declarations in secondary so it will delete the directory even if there is an error while running clean_slate()

awwad commented on demo/demo_primary.py:
```diff
@@ -51,9 +52,12 @@
 CLIENT_DIRECTORY_PREFIX = 'temp_primary'
-client_directory = None
+CLIENT_DIRECTORY = None
+TEMP_PINNED_FILE = ''
```
None is a better value to use, since this has no meaningful value until it is populated at some later point. In any case, I think it's better to avoid additional globals, and here the entire purpose of this new global, `TEMP_PINNED_FILE`, is to clean up a temp file (unlike `CLIENT_DIRECTORY`). This is demo code, not part of the reference implementation, so it's not as important, but for good habits' sake: I would have avoided this variable and instead in the code immediately before the creation of the file (when the filename is known), used atexit.register() to register a call to a function that simply deletes the file whose filename is passed in.

awwad replied:
This variable is still in demo_secondary. I'll just delete it there, too.

awwad commented on demo/demo_primary.py:
```diff
@@ -93,24 +97,26 @@ def clean_slate(
   global _ecu_serial
   global listener_thread
   global use_can_interface
+  global TEMP_DIRECTORY
```
You do not use this; it should be removed.

awwad commented on demo/demo_primary.py:
```diff
@@ -121,14 +127,14 @@ def clean_slate(
-
-
+  
+  
```
Try to avoid trailing whitespace.

awwad commented on demo/demo_primary.py:
```diff
@@ -93,24 +97,26 @@ def clean_slate(
   global listener_thread
   global use_can_interface
+  global TEMP_DIRECTORY
+  atexit.register(clean_up) # To delete the temp pinned file and folder after
```
I'd rather not have this happen in top-level module code, where it will trigger when imported. Better to have this happen immediately before the creation of the directory, which still means you can be sure that it'll be deleted even if an error occurs.

The same goes for demo_secondary.

awwad commented on demo/demo_secondary.py:
```diff
@@ -78,8 +80,10 @@ def clean_slate(
-  global client_directory
+  global CLIENT_DIRECTORY
   global attacks_detected
+  atexit.register(clean_up) # To delete the temp pinned file and folder
```
As noted for the other module, this shouldn't happen in module-level code that would be executed on import.

awwad reviewed (changes requested):
Thanks for the PR! A few changes would help - see comments.

shikhar394 committed:
Made suggested changes to the demo_primary

- Removed the temp_ vairable
- Split up the clean up function into two: one for removing the folder and the other for the file (Pinned.json)
- Changed the position where the atexit function registers the clean up function

shikhar394 committed:
Made suggested changes to the demo_secondary

- Removed the temp_ vairable
- Split up the clean up function into two: one for removing the folder and the other for the file (Pinned.json)
- Changed the position where the atexit function registers the clean up function

shikhar394 committed:
Fixed some spaces and positioning of functions in the code

shikhar394 committed:
Removed global variable for file name from demo_primary

awwad commented:
Partial fix to #24. (Handles demo Primary & Secondary temp files; will consider another PR for the server-side files, which may be useful to keep around anyway)

awwad committed:
Remove unused var in demo_secondary (stored temp fname for deletion)

awwad committed:
In demo, schedule temp file cleanup earlier

just before temp file/folder creation so that the cleanup still occurs if
an error occurs in file/folder creation. Also clarify comments about the
demo's temp file/folder deletion and remove some trailing whitespace.

awwad commented:
Code LGTM now; temp file deletion seems to work; demo still runs correctly on it; tests passing. Merging.
